### DISCUSSION TITLE
Can not create a new adapter

### DISCUSSION THREAD

**Discussion by waynelao** [apache/streampipes#1594] [Q&A] [answered]
Hi there,

I am currently using StreamPipes 0.91.0 and run docker-compose up -d to start the application. I follow the documentation here (https://streampipes.apache.org/docs/docs/use-connect.html) to create a new adapter. Once I fill the required infos (access mode, boker hostname, port) and there is an error on Topic. 
![adapter-issue](https://github.com/apache/streampipes/assets/34040970/dcc97dfb-d031-4943-a868-0a0429b57748)

Any solution to this issue? Really appreciate the help!

**Comment by tenthe** [comment]
Hi @waynelao,
thanks for reporting the problem.

What does the error message say when you click Details?
It seems that you cannot connect to your Kafka cluster. 
In this stage StreamPipes tries to connect to Kafka and read all available topics.
Is Kafka port 9092 configured in Kafka for this inbound IP?

**Reply by waynelao** [reply]
Hi @tenthe,
Thank you for the quick reply. 

Here is the detail error message
![error-info](https://github.com/apache/streampipes/assets/34040970/82dd5853-5cd1-4344-a17f-560d09496e85)
I don't think Kafka is configured in port 9092 as I check the docker.
Here is the docker info:
![docker-info](https://github.com/apache/streampipes/assets/34040970/1e68cf4f-ce39-4e31-a1c6-af95e6b83f35)

Should I manually set up Kafka in port 9092? (I thought by running docker-compose up -d, Kafka is automatically configured).

Thank you!

**Reply by tenthe** [reply]
Ah ok, if you are using the internal Kafka you must use the DNS name "Kafka" instead of the IP address. 

If you want to use the IP address of your host you must change the configurations of Kafka. Please let me know if this fixes your problem.

**Reply by waynelao** [reply]
> Ah ok, if you are using the internal Kafka you must use the DNS name "Kafka" instead of the IP address.

Sorry, I don't really understand. Are you talking about making changes in docker-compose.yml file or something else?


Could you please provide step-by-step guidance on how to make StreamPipes connect to Kafka?
Thank you!

**Reply by tenthe** [reply, marked as answer]
Hi @waynelao,

sure. Here is an example for the configuration:
- **Access Mode:** Unauthenticated Plain
- **Broker Hostname:** kafka
- **Broker Port:** 9092

**Reply by waynelao** [reply]
It works great, thank you!

**Reply by tenthe** [reply]
Great, I am happy to hear that. If you have any problems or feedback feel free to share it in a discussion.
